test/src/main/java/org/axonframework/test/matchers/DeepEqualsMatcher.java

```diff
@@ -83,7 +83,7 @@ public boolean matches(Object actual) {
         if (Objects.equals(expected, actual)) {
             return true;
         }
-        if (hasEqualsMethod(actual.getClass())) {
+        if (hasEqualsMethod(actual.getClass()) && (!containsIgnoreFieldFilter())) {
```

Comment:
I think we'll hit more cases if we check if the `filter` is the default `AllFieldsFilter`. If this is the case, apparently nobody set the filter themselves, or if they set it to `AllFieldsFilter`, they don't care about the contents.